### PR TITLE
AdSupply adapter

### PR DESCRIPTION
**Pull request by msadsupply**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: 'adsupply',
  params: {
    clientId: 'b367CB268B1094004A3689751E7AC568F',
    siteId: '9e3ba26d-eaac-4004-9c7d-c0ac269d0cf2',
    zoneId: 28384,
    endpointUrl: 'engine.4dsply.com'
 }
}
```
- maxim@adsupply.com
- [x] official adapter submission

### CODE REVIEW

File: test/spec/adapters/adsupply_spec.js [hunk 1]
```diff
@@ -0,0 +1,359 @@
+describe('adsupply adapter tests', function () {
+
+	const expect = require('chai').expect;
+	const assert = require('chai').assert;
```
Comment (matthewlane) [inline test/spec/adapters/adsupply_spec.js]: This isn't used, can be removed

File: src/adapters/adsupply.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+var utils = require('../utils');
+const ADSUPPLY_CODE = 'adsupply';
+
+var AdSupplyAdapter = function AdSupplyAdapter() {
+  function _validateParams(params) {
+    if (!params || !params.siteId || !params.zoneId || !params.endpointUrl || !params.clientId) {
+      return false;
+    }
+
+    if (typeof params.zoneId !== "number" || params.zoneId <= 0) {
```
Comment (matthewlane) [inline src/adapters/adsupply.js]: Upcoming ESLint error: prefer single quotes around `number`

File: src/adapters/adsupply.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    var clientId = bidRequest.params.clientId;
+    var zoneProp = 'b' + bidRequest.params.zoneId;
+
+    if (!window[clientId] || !window[clientId][zoneProp]) return;
+
+    var media = window[clientId][zoneProp].Media;
+
+    if (!media) return;
+
+    if (!media.Url || !media.Ecpm || typeof media.Ecpm !== "number" || media.Ecpm <= 0) {
```
Comment (matthewlane) [inline src/adapters/adsupply.js]: Also prefer single quotes around `number` here

File: src/adapters/adsupply.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      adloader.loadScript(requestUrl, _makeCallBackHandler(bid.bidId));
+    }
+  }
+
+  return {
+    callBids: _callBids
+  };
+};
+
+module.exports = AdSupplyAdapter;
```
Comment (matthewlane) [inline src/adapters/adsupply.js]: ESLint error: newline required at end of file

File: test/spec/adapters/adsupply_spec.js (hunk 1 above)
Comment (msadsupply) [inline test/spec/adapters/adsupply_spec.js]: const assert = require('chai').assert;    - REMOVED


const expect = require('chai').expect; - USED in the first test

File: src/adapters/adsupply.js (hunk 4 above)
Comment (msadsupply) [inline src/adapters/adsupply.js]: Added a new line at the end. "Files changed" tab doesn't show it for some reason but if you try to edit the file with online editor you will see there's an empty line at the end of the file. Hope it helps to fix the ESLint error.

File: src/adapters/adsupply.js (hunk 2 above)
Comment (msadsupply) [inline src/adapters/adsupply.js]: Fixed

File: src/adapters/adsupply.js (hunk 3 above)
Comment (msadsupply) [inline src/adapters/adsupply.js]: Fixed

**Pull request merged**